src/main/java/edu/ie3/datamodel/io/source/DataSource.java

```diff
@@ -13,5 +13,6 @@
 public interface DataSource {
 
   /** Creates a stream of maps that represent the rows in the database */
-  Stream<Map<String, String>> getSourceData(Class<? extends UniqueEntity> entityClass);
+  Stream<Map<String, String>> getSourceData(
+      Class<? extends UniqueEntity> entityClass, SourceValidator validateSource);
```

Comment:
I'm thinking out loud here, but maybe it'd be better to have validation happen separately from creating the field mappings? It doesn't really fit here, logically.


I know, the price to pay here would be to read out the source file twice (once only for the headline, second time for the content), but honestly I think that's still a cheap price if we can keep the validation less entangled from regular factory functions here.